### PR TITLE
Remove duplicate loss registration

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:

- loss functional values are saved twice for each step with AWS Pytorch.
- This happens because functional losses are saved by default by the post_hook_for_loss_functional fn in AWS Pytorch.

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by NihalHarish** [comment]
Closing till I find an optimal solution
